src/components/AnalysisPage/ThicknessAnalysisBox/ThickAnalysisBox.jsx

```diff
@@ -89,6 +89,16 @@ const ThickAnalysisBox = ({
     return () => clearTimeout(timer);
   }, [partId]);
 
+  const isDisabled = units === null;
+
+  let tooltipTitle = "";
+
+  console.log(isDisabled);
+  if (isDisabled) {
+    tooltipTitle =
+      "Select units of measurement before entering a threshold value";
+  }
```

Comment:
Should this be in a useEffect or something rather than just floating in the component? I'm guessing that this will run for every render or something rather than just when the units change but I don't really know how react works. 

Also is there a way that this logic could be put into the threhsold box component rather than in the ThickAnalysisBox? I dont know which one would be better so I'm not currently implying that you should do that but im just curious if you considered that too

Reply:
Yeah that makes sense! i can put it in a useEffect with partId and isDisabled as dependencies. Then create a 
`const [tooltipTitle, setTooltipTitle] = useState("");` state and set the state in the useEffect.

To put it in the threshold box component (decimalInput.jsx), I'll just have to pass down units and partId to it as a prop so we it will always know if there are units selected and when a part changes.